src/backend/utils/vkbasalt.py

```diff
@@ -0,0 +1,181 @@
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from os import path, environ, system, remove
+from sys import stderr, exit
+from shutil import copyfile
+
+def parse(args):
+    # Apply default settings if possible
+    if args.default:
+        install_paths = [
+            "/usr/lib/extensions/vulkan/vkBasalt/etc/vkBasalt",
+            "/usr/local",
+            "/usr/share/vkBasalt"
+        ]
+        for i in range(len(install_paths)):
+            if path.exists(path.join(install_paths[i], "vkBasalt.conf")):
+                if args.output:
+                    copyfile(path.join(install_paths[i], "vkBasalt.conf"), path.join(args.output, "vkBasalt.conf"))
+                if args.exec:
+                    environ["ENABLE_VKBASALT"] = "1"
+                    environ["VKBASALT_CONFIG_FILE"] = path.join(install_paths[i], "vkBasalt.conf")
+                    system(f"{args.exec}")
+                return
+        stderr.write("No such path for vkBasalt exists\n")
+        exit(1)
+
+    # Generate config and check for errors
+    if args.effects:
+        file = []
+
+        # --disable-on-launch
+        file.append("enableOnLaunch = ")
+        if args.disable_on_launch:
+            file.append("False\n")
+        else:
+            file.append("True\n")
+
+        # --toggle-key
+        file.append("toggleKey = ")
+        if args.toggle_key:
+            file.append(f"{args.toggle_key[0]}\n")
+        else:
+            file.append("Home\n")
+
+        # --cas-sharpness
+        if args.cas_sharpness:
+            if -1 <= args.cas_sharpness <= 1:
+                file.append(f"casSharpness = {args.cas_sharpness}\n")
+            else:
+                stderr.write("Error: CAS sharpness must be above -1 and below 1\n")
+                exit(1)
+
+        # --dls-sharpness
+        if args.dls_sharpness:
+            if 0 <= args.dls_sharpness <= 1:
+                file.append(f"dlsSharpness = {args.dls_sharpness}\n")
+            else:
+                stderr.write("Error: DLS sharpness must be above 0 and below 1\n")
+                exit(1)
+
+        # --dls-denoise
+        if args.dls_denoise:
+            if 0 <= args.dls_denoise <= 1:
+                file.append(f"dlsDenoise = {args.dls_denoise}\n")
+            else:
+                stderr.write("Error: DLS denoise must be above 0 and below 1\n")
```

Comment:
Warning and error handling in Bottles is done via the logging library, i.e. `logging.error(f"Error: DLS denoise must be above 0 and below 1")`.